src/optimization/model_utils.py

```
@@ -0,0 +1,250 @@
+            "metric": metric,
+            "mode": mode,
+            "verbose": 1,
+            "raise_on_failed_trial": True,
+        }
+        if search_alg is not None:
+            run_kwargs["search_alg"] = search_alg
+        analysis = tune.run(train_fn, **run_kwargs)
+    except Exception as exc:
+        if "fail" in str(exc):
+            raise exc
+        raise RuntimeError(str(exc))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)